Modules/posixmodule.c

```diff
@@ -8196,11 +8196,20 @@ os_preadv_impl(PyObject *module, int fd, PyObject *buffers, Py_off_t offset,
     }
 
     cnt = PySequence_Size(buffers);
-    if (cnt < 0)
+    if (cnt < 0){
         return -1;
+    }
 
-    if (iov_setup(&iov, &buf, buffers, cnt, PyBUF_WRITABLE) < 0)
+#ifndef HAVE_PREADV2
+    if(flags != 0){
```

Comment:
nitpick: add spaces (see PEP 7): "if (flags != 0) {".

Reply:
Done in 47f65b5